### ISSUE TITLE
some problems when combining this project with the modified PX4 firmware"ethzasl_mav_px4"

### ISSUE THREAD

**Issue by sushiye** [ethz-asl/mav_control_rw#43]
hello, in the readme note it says that this project can work with PX4 firmware,  i have tried and succeeded in the process. but i have got a problem that  quadrotors with F330 model can't hover stably. the roll_pitch_yawrate_thrust calculated by the linear mpc was published to the modified mavros topic /mavros/setpoint_raw/roll_pitch_yawrate_thrust, but it changes too quick so the quadrotors can't hover. i think the reason may be that the model used in the linear mpc doesn't match with the PX4 quadrotors model or the roll_pitch_yawrate_thrust is published  to the mavros topic too frequently. could you please give me some advice on how to make the  quadrotors with F330 hover stably? thanks a lot!

**Comment by WangJintao1998** [comment]
> hello, in the readme note it says that this project can work with PX4 firmware, i have tried and succeeded in the process. but i have got a problem that quadrotors with F330 model can't hover stably. the roll_pitch_yawrate_thrust calculated by the linear mpc was published to the modified mavros topic /mavros/setpoint_raw/roll_pitch_yawrate_thrust, but it changes too quick so the quadrotors can't hover. i think the reason may be that the model used in the linear mpc doesn't match with the PX4 quadrotors model or the roll_pitch_yawrate_thrust is published to the mavros topic too frequently. could you please give me some advice on how to make the quadrotors with F330 hover stably? thanks a lot!

I had the same problem with iris before but when I modified the model parameters, it can hover stably. So maybe you could try to modify the model patameters to match your quadrotors.

**Comment by sushiye** [comment]
@WangJintao1998 hello, sorry for my late reply. I have been working on other projects for a long time, and recently I'm back for this project. 
First, the reply for your question "how to work with the PX4 firmware": I publish the desired attitude to the modified PX4 firmware with the modified mavros, both of which are the "ethz-asl" provided. 
Then, I wonder if you could tell me how you get the model parameters or give me your iris model parameters for reference? Because I use the iris model in this project,too.
PS: are you Chinese? I think we have some common points in our research and I wonder if we could talk more by WeChat or other messaging App.

> > hello, in the readme note it says that this project can work with PX4 firmware, i have tried and succeeded in the process. but i have got a problem that quadrotors with F330 model can't hover stably. the roll_pitch_yawrate_thrust calculated by the linear mpc was published to the modified mavros topic /mavros/setpoint_raw/roll_pitch_yawrate_thrust, but it changes too quick so the quadrotors can't hover. i think the reason may be that the model used in the linear mpc doesn't match with the PX4 quadrotors model or the roll_pitch_yawrate_thrust is published to the mavros topic too frequently. could you please give me some advice on how to make the quadrotors with F330 hover stably? thanks a lot!
> 
> I had the same problem with iris before but when I modified the model parameters, it can hover stably. So maybe you could try to modify the model patameters to match your quadrotors.

**Comment by WangJintao1998** [comment]
Yes I am Chinese and of course we can talk on WeChat. As for the iris parameters, I look up into the iris.sdf file.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年8月25日(星期三) 中午1:27
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ethz-asl/mav_control_rw] some problems when combining this project with the modified PX4 firmware&quot;ethzasl_mav_px4&quot; (#43)





 
@WangJintao1998 hello, sorry for my late reply. I have been working on other projects for a long time, and recently i'm back for this project.
 First, the reply for your question "how to work with the PX4 firmware": I publish the desired attitude to the modified PX4 firmware with the modified mavros, both of which are the "ethz-asl" provided.
 Then, I wonder if you could tell me how you get the model parameters or give me your iris model parameters for reference? Because I use the iris model in this project,too.
 PS: are you Chinese? I think we have some common points in our research and I wonder if we could talk more by WeChat or other messaging App.
   
hello, in the readme note it says that this project can work with PX4 firmware, i have tried and succeeded in the process. but i have got a problem that quadrotors with F330 model can't hover stably. the roll_pitch_yawrate_thrust calculated by the linear mpc was published to the modified mavros topic /mavros/setpoint_raw/roll_pitch_yawrate_thrust, but it changes too quick so the quadrotors can't hover. i think the reason may be that the model used in the linear mpc doesn't match with the PX4 quadrotors model or the roll_pitch_yawrate_thrust is published to the mavros topic too frequently. could you please give me some advice on how to make the quadrotors with F330 hover stably? thanks a lot!
  
I had the same problem with iris before but when I modified the model parameters, it can hover stably. So maybe you could try to modify the model patameters to match your quadrotors.
  
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by sushiye** [comment]
@WangJintao1998 谢谢你的回复。我的邮箱是irisanthony@foxmail.com, 请你把微信号发我邮箱，我来添加你！（因为在github上发微信号不太安全）

**Comment by OP-Follower** [comment]
> hello, in the readme note it says that this project can work with PX4 firmware, i have tried and succeeded in the process. but i have got a problem that quadrotors with F330 model can't hover stably. the roll_pitch_yawrate_thrust calculated by the linear mpc was published to the modified mavros topic /mavros/setpoint_raw/roll_pitch_yawrate_thrust, but it changes too quick so the quadrotors can't hover. i think the reason may be that the model used in the linear mpc doesn't match with the PX4 quadrotors model or the roll_pitch_yawrate_thrust is published to the mavros topic too frequently. could you please give me some advice on how to make the quadrotors with F330 hover stably? thanks a lot!

你好！请问您的问题解决了吗？我刚刚接触这个领域，现在要做的是把ethz的这个lmpc接入PX4去做一些测试，但刚刚入门有不少问题，请问可以加您微信或者其他方式请教下吗？（邮箱：mr__ge@163.com，期待您的回复！）

**Comment by Mercurialzz** [comment]
> Yes I am Chinese and of course we can talk on WeChat. As for the iris parameters, I look up into the iris.sdf file.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2021年8月25日(星期三) 中午1:27 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [ethz-asl/mav_control_rw] some problems when combining this project with the modified PX4 firmware&quot;ethzasl_mav_px4&quot; (#43) @WangJintao1998 hello, sorry for my late reply. I have been working on other projects for a long time, and recently i'm back for this project. First, the reply for your question "how to work with the PX4 firmware": I publish the desired attitude to the modified PX4 firmware with the modified mavros, both of which are the "ethz-asl" provided. Then, I wonder if you could tell me how you get the model parameters or give me your iris model parameters for reference? Because I use the iris model in this project,too. PS: are you Chinese? I think we have some common points in our research and I wonder if we could talk more by WeChat or other messaging App. hello, in the readme note it says that this project can work with PX4 firmware, i have tried and succeeded in the process. but i have got a problem that quadrotors with F330 model can't hover stably. the roll_pitch_yawrate_thrust calculated by the linear mpc was published to the modified mavros topic /mavros/setpoint_raw/roll_pitch_yawrate_thrust, but it changes too quick so the quadrotors can't hover. i think the reason may be that the model used in the linear mpc doesn't match with the PX4 quadrotors model or the roll_pitch_yawrate_thrust is published to the mavros topic too frequently. could you please give me some advice on how to make the quadrotors with F330 hover stably? thanks a lot! I had the same problem with iris before but when I modified the model parameters, it can hover stably. So maybe you could try to modify the model patameters to match your quadrotors. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

你好! 我最近也在将该项目部署在PX4上，目前遇到了一些问题，大概来讲是在mavros发布roll_pitch_yawrate_thrust话题后，PX4无法识别yaw_rate的输入从而对yaw进行控制，pitch和roll是正常的（PX4固件和mavros都是该项目下建议使用的），请问可以加您联系方式请教一下嘛，十分感谢（我的邮箱是zzzzk000@foxmail.com）
